### PR TITLE
upgrade commons-io due to CVE-2021-29425

### PR DESCRIPTION
**Pull request by alexanderkjall**
# Description

Upgrade commons-io to 2.8.0, it's a vulnerability in FileNameUtils.normalize that isn't used in this project.

More information about the vulnerability can be found here: https://issues.apache.org/jira/browse/IO-556

I have run the tests locally (which failed to to unrelated javadoc warnings) and done a basic check in my application that uses github-api, but not done any other testing.

### CODE REVIEW

**Review by bitwiseman — CHANGES_REQUESTED** [review]
This project is used by other projects that have pinned a lower version of `commons-io`. 
I think it makes sense for use to upgrade the version we use by default but we need to enforce only using methods that are compatible with lower versions.  Please add a test to [ArchTests.java](https://github.com/hub4j/github-api/blob/main/src/test/java/org/kohsuke/github/ArchTests.java) that checks that we are only using a specified approved list for methods from `commons-io`.

**Comment by bitwiseman** [comment]
As noted, this library does not use the method effected by this CVE so it is not urgent to merge and release.

**Comment by alexanderkjall** [comment]
I looked a bit on ArchTests and to be honest I don't understand how to write a unit test that would fail if a function that's not in the library version 2.4 is used?

**Comment by bitwiseman** [comment]
@alexanderkjall 
It's easy! You just... 
Wait, no, I mean, ArchUnit is pretty friendly, you can...

Oh, well, it looks like the test I wrote to enforce using `assertThat()` during tests was wrong. 

Also, checking the methods is not at all intuitive and the documentation for this scenario is not helpful.  

So, thanks for asking. I've implemented a minimal test that does not ensure non-breakage, but will force people to consider when that add commons method calls.

**Review by bitwiseman — APPROVED** [review]

**Comment by bitwiseman** [comment]
@gsmet @alexanderkjall 
Could you review these changes?

File: src/test/java/org/kohsuke/github/ArchTests.java [hunk 1]
```diff
@@ -113,14 +138,103 @@ public void testBetaApisAreFlaggedAsDeprecated() {
...
                 .because(reason);
 
-        onlyAssertThatRule.check(tesetClassFiles);
+        onlyAssertThatRule.check(testClassFiles);
+    }
+
+    @Test
+    public void testRequireUseOfOnlySpecificApacheCommons() {
+
+        final ArchRule onlyApprovedApacheCommonsLang3Methods = classes()
```
Comment (gsmet) [inline src/test/java/org/kohsuke/github/ArchTests.java:156]:
```suggestion
        final ArchRule onlyApprovedApacheCommonsMethods = classes()
```
`IOUtils` is in commons-io.

File: src/test/java/org/kohsuke/github/ArchTests.java [hunk 2]
```diff
@@ -113,14 +138,103 @@ public void testBetaApisAreFlaggedAsDeprecated() {
...
+                                Object.class,
+                                Boolean.class),
+                        targetMethodIs(ReflectionToStringBuilder.class, "accept", Field.class),
+                        targetMethodIs(IOUtils.class, "closeQuietly", InputStream.class),
+                        targetMethodIs(IOUtils.class, "closeQuietly", Closeable.class),
+                        targetMethodIs(IOUtils.class, "toString", InputStream.class, Charset.class),
+                        targetMethodIs(IOUtils.class, "toString", Reader.class),
+                        targetMethodIs(IOUtils.class, "toByteArray", InputStream.class)))
+                .because(
+                        "Commons methods must be manually verified to be compatible with commons-io:2.4 or earlier and commons-lang3:3.9 or earlier should be used.");
```
Comment (gsmet) [inline src/test/java/org/kohsuke/github/ArchTests.java:190]:
```suggestion
                        "Commons methods must be manually verified to be compatible with commons-io:2.4 or earlier and commons-lang3:3.9 or earlier.");
```

File: src/test/java/org/kohsuke/github/ArchTests.java [hunk 3]
```diff
@@ -113,14 +138,103 @@ public void testBetaApisAreFlaggedAsDeprecated() {
...
+
+        onlyApprovedApacheCommonsLang3Methods.check(classFiles);
```
Comment (gsmet) [inline src/test/java/org/kohsuke/github/ArchTests.java:192]:
```suggestion
        onlyApprovedApacheCommonsMethods.check(classFiles);
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Spotted a few minor things.

I'm not very familiar with ArchUnit but I suppose you have tested adding another method fails the build?

**Comment by bitwiseman** [comment]
@gsmet  Yes, I've verified that methods not on the list are caught by the test.

**Comment by jeffalder** [comment]
@bitwiseman - Is it possible to get a release cut that includes this PR? I'd be interested in upgrading to make the CVE go away.

**Comment by bitwiseman** [comment]
@jeffalder Done.

**Pull request merged**